packages/wallet/contract/src/walletFactory.js

```diff
@@ -34,12 +44,32 @@ export const start = async (zcf, privateArgs) => {
   assert(namesByAddress, 'missing namesByAddress');
   assert(agoricNames, 'missing agoricNames');
   const zoe = zcf.getZoeService();
-  const { storageNode } = privateArgs;
+  const { storageNode, bridgeManager } = privateArgs;
 
   /** @type {MapStore<string, import('./smartWallet').SmartWallet>} */
   const walletsByAddress = makeScalarBigMapStore('walletsByAddress');
   const provider = makeAtomicProvider(walletsByAddress);
 
+  const handleWalletAction = Far('walletActionHandler', {
+    /**
+     *
+     * @param {string} srcID
+     * @param {WalletAction} obj
+     */
+    fromBridge: async (srcID, obj) => {
```

Comment:
Just thinking out loud, we will at some point want `provideSmartWallet` to be callable in this same way so a user can create their wallet from the browser.

Reply:
For follow-up, see notes in #4654